tests/rep.py

```diff
@@ -71,15 +76,88 @@ def test_single_errors_rep(self):
 
                 # Simulate
                 results = (
-                    execute(
-                        qubit.circ,
-                        Aer.get_backend("aer_simulator"),
-                        shots=1000,
-                    )
+                    execute(qubit.circ, Aer.get_backend("aer_simulator"), shots=1000,)
+                    .result()
+                    .get_counts()
+                )
+
+                # Get Logical Error Rate
+                logical_error_rate = self.get_logical_error_rate(results, 0)
+                print(
+                    f"Decoding result for {error} Error on the {i}th data qubit, logical_error_rate: {logical_error_rate}."
+                )
+                self.assertEqual(
+                    logical_error_rate,
+                    0,
+                    f"Decoding did not work for an {error} Error on the {i}th data qubit.",
+                )
+
+
+class TestPhaseFlipProtectedRep(unittest.TestCase):
+    """
+    Test the Repetition Rotated Surface Code
+    """
+
+    def setUp(self):
+        self.params = {"d": 5, "phase-flip-protected": True}
+        self.params["T"] = 1
+        self.decoder = RepetitionDecoder(self.params)
+
+    def get_logical_error_rate(
+        self, readout_strings, correct_logical_value, err_prob=None
+    ):
+        """
+        Args:
+            readout_strings: a dictionary of readout strings along with counts
+            e.g. {"1 00000000 00000000":48, "1 00100000 00100000":12, ...} in the case of d=3, T=2
+
+            correct_logical_value: integer (0/1) depicting original encoded logical value
+
+        Returns:
+            error_rate:
+                float = (# of unsuccessful logical value predictions) / (total # of pred )
+        """
+        total_count = 0
+        total_errors = 0
+        for readout, count in readout_strings.items():
+            total_count += count
+            predicted_logical_value = self.decoder.correct_readout(
+                readout, "Z", err_prob=err_prob
+            )
+            if predicted_logical_value != correct_logical_value:
+                total_errors += count
+
+        return total_errors / total_count
+
+    def test_single_errors_rep(self):
+        """
+        Setting up a |+z> state, stabilizing twice, and reading out logical Z.
+        Inserting Z gates on each data qubit between tbe two stabilizer measurement rounds.
+
+        Then, testing:
+        1.  The MWPM decoder is able to correct these single qubit errors
+            and predict the correct logical readout value.
+        """
+        # set up circuit
+        d = self.params["d"]
+        for i in range(d[constants.DH] * d[constants.DW]):
+            for error in ["z"]:
+                # Set up circuit
+                qubit = RepetitionQubit(self.params)
+                qubit.reset_z()
```

Comment:
So, while you're inserting the correct error (a phase flip Z), this should use `reset_x` and `readout_x`, if  you want to run a meaningful test. 

If you initialize logical |+z>, insert a Z error, and then do logical Z readout, you will always get 0 (even without any decoding). 

This can be understood by considering a single qubit state in |+z>. Clearly, a Z flip (Z|+z> = |+z>) will not affect the single qubit state |+z> since it's already an eigenstate of Z.

Reply:
Yeah, that was my thought, which is why I mentioned in my PR description. How come we don't perform tests this way for the XXZZ tests though?

Reply:
Ah, I missed that. You're absolutely right, we should do that for XXZZ as well.
This makes me think our XZZX tests may also be off. 
Let's fix these repetition code tests first and address this new #71 issue in a later PR.